lib/pdk/util/git.rb

```diff
@@ -81,10 +90,12 @@ def self.remote_repo?(maybe_repo)
 
       def self.work_tree?(path)
         return false unless PDK::Util::Filesystem.directory?(path)
+        result = cached_git_query(path, :work_tree?)
+        return result unless result.nil?
```

Comment:
Is it too much optimization to check the cache before hitting the filesystem with `.directory?` here?

Reply:
I felt it was. and as per the workflow above, if the directory is _later_ created then we'd get false cache hits.